cfgov/jinja2/v1/_layouts/base.html

```diff
@@ -318,7 +318,11 @@
           #}
           {% if page and page.media_js %}
             {% for js in page.media_js %}
-              s.push( '{{ static('js/routes/on-demand/' + js) }}' );
+                {% if 'https://' in js %}
+                  s.push('{{ js}}');
+                {% else %}
+                  s.push( '{{ static('js/routes/on-demand/' + js) }}' );
+                {% endif %}
```

Comment:
I defer to @chosak, but would it make any sense to attempt move this logic out of the template? I notice that you can have logic in some of these JS media lists, like is shown here https://docs.djangoproject.com/en/4.1/topics/forms/media/#paths-as-objects

That could maybe bury it, and we do have if/else logic around this code block already. I'm just wondering if this comes up for anything other than the Youtube API? If it's only Youtube, maybe we run the logic in that context only and not have every on-demand script checked for `https://`.

Reply:
I think your instinct is good here (not running unnecessary logic in the templates where possible) but I think the logic to actually pull out, say, an `external_js` list would be far more expensive than this check and would have to be run everywhere (since base.html can't assume what does and doesn't have `external_js`

Something we COULD do is convert the media_js files to their static location in python, but that would require updating all the `Media` classes for everything that includes js this way

Reply:
I do agree that, in principle, it'd be nice for this logic to live outside the templates and be consolidated into some larger static file handling.

However, absent a larger need than for this one example, I think this is fine.